src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadWidget.java

```diff
@@ -104,17 +113,68 @@ public void onSuccess(DiscussionThreadBundle result) {
 
 	@Override
 	public void toggleThread() {
+		if (view.isThreadCollapsed()) {
+			// expand
+			view.setThreadDownIconVisible(false);
+			view.setThreadUpIconVisible(true);
+			configureMessage();
+		} else {
+			// collapse
+			view.setThreadDownIconVisible(true);
+			view.setThreadUpIconVisible(false);
+		}
 		view.toggleThread();
 	}
 
+	public void configureMessage() {
+		synAlert.clear();
+		discussionForumClientAsync.getThreadUrl(threadId, new AsyncCallback<MessageURL>(){
+
+			@Override
+			public void onFailure(Throwable caught) {
+				synAlert.handleException(caught);
+			}
+
+			@Override
+			public void onSuccess(MessageURL result) {
+				requestBuilder.configure(RequestBuilder.GET, result.getMessageUrl());
```

Comment:
For code clarity, I suggest refactoring this into it's own method (it takes the result.getMessageUrl() String as input).